pkg/operator/k8sutil/k8sutil.go

```diff
@@ -215,3 +215,78 @@ func UsePDBV1Beta1Version(Clientset kubernetes.Interface) (bool, error) {
 	minVersionForPDBV1 := "1.21.0"
 	return k8sVersion.LessThan(version.MustParseSemantic(minVersionForPDBV1)), nil
 }
+
+// ToValidDNSLabel converts a given string to a valid DNS-1035 spec label. The DNS-1035 spec
+// follows the regex '[a-z]([-a-z0-9]*[a-z0-9])?' and is at most 63 chars long. DNS-1035 is used
+// over DNS-1123 because it is more strict. Kubernetes docs are not always clear when a DNS_LABEL is
+// supposed to be 1035 or 1123 compliant, so we use the more strict version for ease of use.
+//   - Any input symbol that is not valid is converted to a dash ('-').
+//   - Multiple resultant dashes in a row are compressed to a single dash.
+//   - If the starting character is a number, a 'd' is prepended to preserve the number.
+//   - Any non-alphanumeric starting or ending characters are removed.
+//   - If the resultant string is longer than the maximum-allowed 63 characters], characters are
+//     removed from the middle and replaced with a double dash ('--') to reduce the string to 63
+//     characters.
+func ToValidDNSLabel(input string) string {
```

Comment:
This helper is just to avoid some error scenarios where the volume name might not be valid? It's a complex method, just wanted to confirm if it's a common case or a corner case that we're solving.

Reply:
It's actually the primary case for what it's used for in this PR. In short, we want to convert any `/`, `_`, `.`, or other special character into a `-` so that the volume name we create doesn't give errors.

Reply:
I see now, to convert the path to a name, and the path will certainly have those chars.